runtime/src/iree/io/formats/gguf/gguf_format_test.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/io/formats/gguf/gguf_format.h"
+
+#include "iree/base/internal/file_io.h"
+#include "iree/io/formats/gguf/testdata/gguf_files.h"
+#include "iree/testing/gtest.h"
+#include "iree/testing/status_matchers.h"
+
+namespace iree {
+namespace {
+
+static iree_io_file_handle_t* OpenTestFile(const char* name) {
+  const struct iree_file_toc_t* file_toc = iree_io_gguf_files_create();
+  for (size_t i = 0; i < iree_io_gguf_files_size(); ++i) {
+    if (strcmp(file_toc[i].name, name) == 0) {
+      iree_io_file_handle_t* file_handle = NULL;
+      IREE_CHECK_OK(iree_io_file_handle_wrap_host_allocation(
+          IREE_IO_FILE_ACCESS_READ,
+          iree_make_byte_span((void*)file_toc[i].data, file_toc[i].size),
+          iree_io_file_handle_release_callback_null(), iree_allocator_system(),
+          &file_handle));
+      return file_handle;
+    }
+  }
+  IREE_CHECK_OK(iree_make_status(
+      IREE_STATUS_NOT_FOUND,
+      "test file `%s` not found embedded into test binary", name));
+  return NULL;
+}
+
+TEST(GgufFormatTest, Empty) {
+  iree_io_parameter_index_t* index = NULL;
+  IREE_ASSERT_OK(
+      iree_io_parameter_index_create(iree_allocator_system(), &index));
+
+  iree_io_file_handle_t* file_handle = OpenTestFile("empty.gguf");
+  IREE_ASSERT_OK(iree_io_parse_gguf_index(file_handle, index));
+  iree_io_file_handle_release(file_handle);
+
+  iree_io_parameter_index_release(index);
+}
+
+TEST(GgufFormatTest, SingleTensors) {
```

Comment:
nit: singular/plural?
```suggestion
TEST(GgufFormatTest, SingleTensor) {
```